api/src/main/java/org/pmiops/workbench/notebooks/NotebooksServiceImpl.java

```diff
@@ -158,16 +160,21 @@ public FileDetail copyNotebook(
   public FileDetail cloneNotebook(
       String workspaceNamespace, String workspaceName, String fromNotebookName) {
     String newName = "Duplicate of " + fromNotebookName;
-    final FileDetail result =
+    final FileDetail copiedNotebookFileDetail =
         copyNotebook(
             workspaceNamespace,
             workspaceName,
             fromNotebookName,
             workspaceNamespace,
             workspaceName,
             newName);
-    monitoringService.recordIncrement(MonitoringViews.NOTEBOOK_CLONE);
-    return result;
+    monitoringService.recordBundle(
+        MeasurementBundle.builder()
+            .addValue(EventMetric.NOTEBOOK_CLONE, 1L)
+            .attach(AttachmentKey.NOTEBOOK_NAME, copiedNotebookFileDetail.getName())
```

Comment:
[side question] I could be wrong, but isn't it best practice not to set attachments / facets on fields with an unbounded space of possible values?

There's also a potential concern here around user data privacy, where the file name of a researcher's notebook may include details about the nature of their research. Permanently attaching this to metrics in Stackdriver, which is otherwise generally used to track operational metrics without an expectation of housing private data, may be a slight risk.

(Musings from past experience: Google is probably exceptionally conservative in this regard, but my past team was required to treat *cloud project names* as PII, subject to extreme auditing & access requirements and wipeout compliance. We're likely a solid step less strict than that, but it's a constant thought in my head.)

Reply:
Quick amendment: based on some unrelated digging into Stackdriver metrics, I'm reasonably convinced that my internalized guidance of "don't use unbounded values for attachment fields" is outdated. They seem pretty happy to store arbitrary strings here; presumably there's some fairly high limit re: the number of independent strings that they like to track during any one time window, but maybe that's not likely something we'll run up against.

Reply:
Further amendment to the privacy implications: due to similarly-unrelated Stackdriver digging, I think I'm seeing that metrics tend to be kept for only 6 weeks by default, unless you actively do something to permanently store them. This makes me less concerned on the privacy front, though this is the kind of thing where I'd want to be sure we have an anticipated use case in mind before we just start tracking it because we can.

Reply:
All very good points. 

I really just needed an event metric with some kind of attribute. It definitely makes sense to only do this when you really want it.

I should definitely think about our retention policy. I was hoping it would save older data at a lower resolution.

Reply:
Dropped the notebook name & workspace properties for now, since they were just placeholders.

I don't yet have a continuous-valued attachment use case, but it should work.